### ISSUE TITLE
transition/transversion scoring matrix in Bio/Phylo/TreeConstruction.py

### ISSUE THREAD

**Issue by mdehoon** [biopython/biopython#2917]
In `Bio/Phylo/TreeConstruction.py`, the transition/transversion scoring matrix is defined in the `DistanceCalculator` class as 
```
    # transition/transversion scoring matrix
    trans = [[6], [-5, 6], [-5, -1, 6], [-1, -5, -5, 6]]
```
so
```
       A       T       C       G
A      6      -5      -5      -1
T     -5       6      -1      -5
C     -5      -1       6      -5
G     -1      -5      -5       6
```
Is there a reference for this transition/transversion scoring matrix? A commonly used transition/transversion scoring matrix is the one defined by David Wheeler ("Weight Matrices for Sequence Similarity Scoring", Version 2.0, May 1996):
```
      A      T      C      G
A     0      5      5      1
T     5      0      1      5
C     5      1      0      5
G     1      5      5      0
```
(defined as a penalty matrix).

**Comment by etal** [comment]
FYI @lijax , the original author of this code -- do you remember the source for this scoring matrix?

**Comment by etal** [comment]
Looks like it initially landed in commit 3087f3b675 from GSoC 2013. I don't see a description of the scoring matrix choices in the [project blog](https://lijax.github.io/en/index2.html). I wonder if some non-obvious math was involved in converting the Wheel matrix to the values we see here, with 6 on the diagonal -- subtracting Wheeler from scalar 6 doesn't do it.

**Comment by mdehoon** [comment]
@etal , @lijax  I guess there is no absolute right or wrong matrix here. If there is no reference for the current matrix, would you be OK if we replace it by the one defined by David Wheeler?

**Comment by lijax** [comment]
hi @mdehoon. The only thing I can remember is that it was from an populoar tool, mega or another command line tool, I forgot which one is. Sorry, better to reference it earlier. Would you like to refactor the class in order to provide custom matrix?

**Comment by mdehoon** [comment]
@lijax  Thank you for the explanation. It's actually because I would like to replace the use of `MatrixInfo` from `Bio.SubsMat` by `substitution_matrices` from `Bio.Align`, since `Bio.SubsMat` is currently not being maintained. The substitution matrices from `Bio.Align` would also be easier to use in `Bio.Phylo.TreeConstruction` as you won't need to go through `_build_protein_matrix` (I think). `Bio.Align.substitution_matrices` already contains David Wheeler's transition/transversion matrix, so if you're OK with that I would like to use that one instead of the current `trans` matrix in  `Bio.Phylo.TreeConstruction`.

**Comment by lijax** [comment]
I'm OK. It would be much cleaner to import it from `Bio.Align.substitution_matrices`. What about you? @etal

**Comment by mdehoon** [comment]
Fixed by #2986
